src/gen_alpha_poc/quaternion.h

```diff
@@ -157,17 +157,18 @@ Vector rotation_vector_from_quaternion(const Quaternion& quaternion) {
     auto q2 = quaternion.GetYComponent();
     auto q3 = quaternion.GetZComponent();
 
-    auto sin_angle_squared = q1 * q1 + q2 * q2 + q3 * q3;
-
-    // Return the rotation vector {0, 0, 0} if provided quaternion is null
-    if (close_to(sin_angle_squared, 0.)) {
-        return Vector{0.0, 0.0, 0.0};
+    Vector n(q1, q2, q3);                           // Vector part of quaternion
+    double n_length = n.Length();                   // magnitude of vector
+    double n_length_squared = n_length * n_length;  // magnitude squared
+
+    // If the length of the vector is less than machine precision,
+    // calculate and return rotation vector using method stable for small angles
+    if (n_length < kEpsilon) {
```

Comment:
Is machine precision the right value here, or do we want to do this at some multiple of kEpsilon? We may want to have some unit tests the exercise this limit to make sure we're switching methods at the right time.

Reply:
I don't have a unit test in this code, but I did pretty significant testing in Python to verify that it was working properly. Since the issue is caused by floating point precision, machine precision seems to be correct here.